### ISSUE TITLE
jsonlite error message

### ISSUE THREAD

**Issue by smartinsightsfromdata** [jrowen/rhandsontable#3]
@jrowen Congrats for this very neat implementation of handsontable!
I haven't tested it fully yet (watch this space!), but in the shiny.R example  I notices the error message below with  (probably after shiny recent adoption of jsonlite):

```R
Input to asJSON(keep_vec_names=TRUE) is a named vector. 
In a future version of jsonlite, this option will not be supported, and named vectors will be translated into arrays instead of objects. 
If you want JSON object output, please use a named list instead. See ?toJSON.
```

**Comment by jrowen** [comment]
I'm not seeing this error with `shiny` v0.11.1.

**Comment by smartinsightsfromdata** [comment]
@jrowen I have v0.11.1.9000 which I believe is the version where they migrated to jsonlite (they are planning soon to do the same with htmlwidgets).

**Comment by smartinsightsfromdata** [comment]
@jrowen correct: it does not happen with ```shiny``` v0.11.1.  Nevertheless as the issue is connected to ```shiny``` migration to ```jsonlite``` will need appropriate tlc...

**Comment by jrowen** [comment]
Can you post the example that generates an error?

**Comment by smartinsightsfromdata** [comment]
This is the code (imperfect - it is just for me to test the insurgence of endless loops):
```R
options(warn=-1)
library(rhandsontable)
library(shiny)

options(warn=-1)
quantity <- id <- 1:20
label <- paste0("lab","-",quantity)
pick <- FALSE
iris_ <- data.frame(id=id,pick=pick, quantity=quantity,label=label,iris[1:20,] ,stringsAsFactors = FALSE)

iris_$Species <- NULL
cols <- colnames(iris_)
#---------------------------
ui <- fluidPage(
	fluidRow(
		column(8,rHandsontableOutput('demTb')),
		column(3,uiOutput("demSli"))
		)
	)

server <- function(session, input, output) {


output$demSli <- renderUI({

if(is.null(input$demTb) ) return()
  
isolate({
df_ <- hot_to_r(input$demTb)
index <- which(df_$pick==T)
if(length(index)==0) return()
labs <- iris_$label[index] 
pages <- "test"
iter <- length(labs)
buttn <- 1
valLabs <- sapply(1:iter, function(i) {
if(is.null(input[[paste0(pages,"d",labs[i],buttn)]] )) {
          0
} else {  as.numeric(input[[paste0(pages,"d",labs[i],buttn)]])  }
}) 
#
toRender <- lapply(1:iter, function(i) {
  sliderInput(inputId = paste0(pages,"d",labs[i],buttn),
              label =  h6(paste0(labs[i],"")),
              min = -100,
              max = 100,
              step = 1,
              value = valLabs[i],
              post="%",
              ticks = FALSE, animate = FALSE)
              })
})
      return(toRender)

})
#--------------------
rds <- reactive({

if( is.null(input$demTb) ) return(iris_)

df_ <- hot_to_r(input$demTb)
isolate({

index <- which(df_$pick==T) 
if(length(index)==0) return(df_)
labs <- iris_$label[index] 
pages <- "test"
iter <- length(labs)
buttn <- 1
}) # end isolate
valLabs <- sapply(1:iter, function(i) {
    if(is.null(input[[paste0(pages,"d",labs[i],buttn)]] )) {
      0
    } else {  
      as.numeric(input[[paste0(pages,"d",labs[i],buttn)]])/100  
    }
  })

  dft_ <- data.frame(label=labs, multi=valLabs, stringsAsFactors = FALSE)
  dft_ <- merge(iris_,dft_,by="label", all.x=T)
  
  dft_$quantity <- sapply(1:length(dft_$quantity), function(z) {
    if( is.na( dft_$multi[z]) ) { 
    dft_$quantity[z]
  } else { iris_$quantity[z]*(1 + dft_$multi[z]) }
})
dft_[with(dft_,order(as.numeric(id))),]
df_[with(df_,order(as.numeric(id))),]

df_$quantity <- df_$quantity
  return(df_)
  }) 


output$demTb  <-  renderRHandsontable({


if(is.null(rds() )) return()

df_ <- rds() 
df_ <- df_[cols]
df_ <- df_[with(df_,order(as.numeric(id))),]
rhandsontable(df_, readOnly = FALSE, rowHeaders= NULL, useTypes= TRUE) %>%
  hot_table(highlightCol = TRUE, highlightRow = TRUE) %>%
hot_validate_numeric(col = "quantity", min = 0, max = 50, exclude = 40)


})

}

shinyApp(ui, server)
```

**Comment by smartinsightsfromdata** [comment]
@jrowen If you could look kindly into this issue - as well as the validator logic issue (by typeof) I would love to test the widget more extensively with the idea to use it in one of my projects.

**Comment by jrowen** [comment]
I'd prefer to make any necessary updates once a new version of `shiny` is stable and released.

**Comment by smartinsightsfromdata** [comment]
@jrowen fair enough - but what about the validator logic issue (by typeof)?

**Comment by smartinsightsfromdata** [comment]
@jrowen Not sure if you have seen this: https://github.com/ramnathv/htmlwidgets/pull/28#issuecomment-91634755
htmlwidget is ready to migrate to jsonlite.  Sadly I'm really busy this weekend and I do not have much time for testing.
These are the packages you need:
```R
devtools::install_github(c('jeroenooms/jsonlite', 'rstudio/shiny', 'yihui/htmlwidgets@jsonlite'))
```

**Comment by smartinsightsfromdata** [comment]
@jrowen I tried the shiny app posted here earlier and I cannot see anything anymore!

**Comment by jrowen** [comment]
I think we have a [fix for the htmlwidgets move to jsonlite](https://github.com/jrowen/rhandsontable/tree/jsonlite).  

The informational message mentioned above still appears when a `data.frame` is passed to the widget.  rhandsontable doesn't set the `keep_vec_names` parameter, so the message may be originating from htmlwidgets or shiny.

**Comment by yihui** [comment]
FYI, I have fixed this issue in my PR #8.
